packages/video_player/video_player/test/video_player_test.dart

```diff
@@ -1204,6 +1205,64 @@ void main() {
     expect(colors.bufferedColor, bufferedColor);
     expect(colors.backgroundColor, backgroundColor);
   });
+
+  test('isCompleted updates on video end', () async {
+    final VideoPlayerController controller = VideoPlayerController.networkUrl(
+      _localhostUri,
+      videoPlayerOptions: VideoPlayerOptions(),
+    );
+
+    await controller.initialize();
+
+    final StreamController<VideoEvent> fakeVideoEventStream =
+        fakeVideoPlayerPlatform.streams[controller.textureId]!;
+
+    bool currentIsCompleted = controller.value.isCompleted;
+
+    final void Function() isCompletedTest = expectAsync0(() {});
+
+    controller.addListener(() async {
+      if (currentIsCompleted != controller.value.isCompleted) {
+        currentIsCompleted = controller.value.isCompleted;
+        if (controller.value.isCompleted) {
+          isCompletedTest();
+        }
+      }
+    });
+
+    fakeVideoEventStream.add(VideoEvent(eventType: VideoEventType.completed));
+  });
+
+  test('isCompleted updates on video end', () async {
+    final VideoPlayerController controller = VideoPlayerController.networkUrl(
+      _localhostUri,
+      videoPlayerOptions: VideoPlayerOptions(),
+    );
+
+    await controller.initialize();
+
+    final StreamController<VideoEvent> fakeVideoEventStream =
+        fakeVideoPlayerPlatform.streams[controller.textureId]!;
+
+    bool currentIsCompleted = controller.value.isCompleted;
+
+    final void Function() isCompletedTest = expectAsync0(() {});
+    final void Function() isNoLongerCompletedTest = expectAsync0(() {});
+    controller.addListener(() async {
+      if (currentIsCompleted != controller.value.isCompleted) {
+        currentIsCompleted = controller.value.isCompleted;
+        if (controller.value.isCompleted) {
+          isCompletedTest();
+          fakeVideoEventStream
+              .add(VideoEvent(eventType: VideoEventType.isPlayingStateUpdate));
+        } else {
+          isNoLongerCompletedTest();
+        }
+      }
+    });
+
+    fakeVideoEventStream.add(VideoEvent(eventType: VideoEventType.completed));
+  });
 }
 
```

Comment:
There should be a test for each case that is supposed to set `isCompleted` back to `false`.